sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/fileSourceInterfaces.scala

```diff
@@ -172,6 +171,13 @@ case class HadoopFsRelation(
 }
 
 /**
+ * A helper class to list up qualified files in parallel.
+ */
+private[spark] abstract class SerializablePathFilter extends PathFilter with Serializable {
```

Comment:
Extending from `PathFilter` makes internal implementation a little bit easier, but I'd prefer to avoid depending on Hadoop classes/interfaces in Spark SQL public interfaces whenever possible.